src/main/java/org/networkcalculus/dnc/network/server_graph/ServerGraph.java

```diff
@@ -757,6 +757,33 @@ public Flow getFlow(int id) throws Exception {
 		return result;
 	}
 
+    /**
+     * Finds all the flows that have exactly path p as their path.
+     * @param p
+     *             The path which all the returned flows have.
+     * @return Set of flows that have p as their path.
+     * @throws Exception
+     * 			   	Could not find although there may be flows.
+     */
+    public Set<Flow> getFlowsByPath(Path p) throws Exception {
+
+        Set<Flow> result = new HashSet<Flow>();;
+
+        if(p != null )
```

Comment:
else?
There is no flow having a null path, i.e., the missing else-clause could simply return the empty results set. However, that might lead to seemingly unexpected behavior that is very hard to debug. I prefer to throw a RuntimeException that breaks the entire execution.

Reply:
As it is right now, if the Path is null then the method will return an empty HashSet. Okay, I will throw a RuntimeException in that case instead.

Reply:
Can you add the according description to the JavaDocs, please?

Reply:
I have added a comment within the method and in the description of the method for the case where the path is null.